pybwa/libbwamem.pyx

```diff
@@ -898,18 +898,18 @@ cdef class BwaMem:
                 rec.reference_id = mem_aln.rid
                 rec.reference_start = mem_aln.pos
                 rec.mapping_quality = mem_aln.mapq
-                cigar = ""
+                cigartuples = []
                 cigar_len_sum = 0
                 for k in range(mem_aln.n_cigar):
                     cigar_op = mem_aln.cigar[k] & 0xf
                     if not opt.soft_clip_supplementary and mem_aln.is_alt == 0 and (
                             cigar_op == 3 or cigar_op == 4):
                         cigar_op = 4 if j > 0 else 3  # // use hard clipping for supplementary alignments
                     cigar_len = mem_aln.cigar[k] >> 4
-                    cigar += f"{cigar_len}" + "MIDSH"[cigar_op]
+                    cigartuples.append((cigar_op, cigar_len))
```

Comment:
Unfortunately the operator values in [`bwa mem`](https://github.com/lh3/bwa/blob/79b230de48c74156f9d3c26795a360fc5a2d5d3b/bwamem.c#L846) are not the same as those in [`pysam`](https://github.com/pysam-developers/pysam/blob/3e3c8b0b5ac066d692e5c720a85d293efc825200/pysam/libcalignedsegment.pyx#L3372-L3382) so this wont work.